### PR TITLE
Additional Testing Around Blacklist and ConfigurationClient

### PR DESCRIPTION
**Pull request by electrostat**
Add additional testing (unit and instrumentation) for Blacklist and ConfigurationClient. Goal is to increase test coverage, handle corner cases better, and prepare for wider array of testing platforms (api levels + devices).

Fixes: https://github.com/mapbox/mapbox-events-android/issues/330

### CODE REVIEW

File: libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java [hunk 1]
```diff
@@ -100,7 +101,9 @@ private static boolean isValidContent(String data) {
     JsonArray jsonArray;
     try {
       JsonObject responseJson = gson.fromJson(data, JsonObject.class);
-      jsonArray = responseJson.getAsJsonArray("RevokedCertKeys");
+      JsonElement jsonElement = responseJson.get("RevokedCertKeys");
```
Comment (andrlee) [inline libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java]: does `responseJson.getAsJsonArray("RevokedCertKeys");` returns `null` if object is not array? so why do we need additional helper on line 140?

File: libtelemetry/src/test/java/com/mapbox/android/telemetry/ConfigurationClientTest.java [hunk 2]
```diff
@@ -26,6 +32,8 @@
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 public class ConfigurationClientTest {
```
Comment (andrlee) [inline libtelemetry/src/test/java/com/mapbox/android/telemetry/ConfigurationClientTest.java]: should we also test for a `null` handler passed into the client?

File: libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    setTimeStamp(System.currentTimeMillis() - DAY_IN_MILLIS);
+    Call mockedCall = mock(Call.class);
+    IOException mockedException = mock(IOException.class);
+
+    configurationClient.onFailure(mockedCall, mockedException);
+
+    assertFalse(configurationClient.shouldUpdate());
+  }
+
+  private void setTimeStamp(Long milliseconds) {
```
Comment (andrlee) [inline libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java]: boxed type?

File: libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.io.IOException;
+
+import okhttp3.Call;
+import okhttp3.OkHttpClient;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class ConfigurationClientInstrumentationTest {
```
Comment (andrlee) [inline libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java]: are we actually hitting real backend with this instrumentation test?

File: libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java (hunk 1 above)
Comment (electrostat) [reply libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java]: During testing, this did not hold true. Test failed when writing check with `IllegalStateException`. Wrote boolean check to correct that issue. Just dug into gson documentation to confirm and that is what it says. It did enlighten me to a method that already existed for this functionality though, so will replace with new method.

https://www.javadoc.io/doc/com.google.code.gson/gson/2.8.5

File: libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java (hunk 4 above)
Comment (electrostat) [reply libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java]: We are currently not hitting real backend. Spent 2+ hours working on solution and decided to get the ball rolling on the review process. Did not find solution to this amongst coworkers and online resources. `CountdownLatch`, which I used previously for similar Async tests is also not working in this situation. 

Please advise or point in right direction for hunting down a good solution on this.

Comment (andrlee) [reply libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/ConfigurationClientInstrumentationTest.java]: would this be helpful: https://developer.android.com/reference/androidx/test/espresso/idling/net/UriIdlingResource ?
